lisa/tools/python.py

```diff
@@ -89,3 +90,66 @@ def exists_package(self, package_name: str) -> bool:
     def uninstall_package(self, package_name: str) -> bool:
         result = self.run(f"uninstall {package_name} -y", force_run=True, sudo=True)
         return result.exit_code == 0
+
+
+class PythonVenv(Python):
+    @property
+    def command(self) -> str:
+        return f"{self.get_venv_path()}/bin/{super().command}"
+
+    def _check_exists(self) -> bool:
+        # _super = type(super())
+        # assert isinstance(super(), Python)
+        return self.node.execute("python3 -m venv --help", shell=True).exit_code == 0
+        # return self.node.execute("python3 -m venv --help").exit_code == 0
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+
+    def _install(self) -> bool:
+        super()._install()
+        if isinstance(self.node.os, Posix):
+            self.node.os.install_packages("python3-venv")
+        return self._check_exists()
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Python]
+
+    # if get_venv_path is invoked before create_venv, it will create
+    # a venv in the node working path
+    def get_venv_path(self) -> PurePath:
+        if not hasattr(self, "_venv_path"):
+            self._venv_path = self.create_venv()
+        return self._venv_path
+
+    def create_venv(self, venv_path: str = "", sudo: bool = False) -> PurePath:
+        _venv_path: PurePath = (
+            PurePath(venv_path) if venv_path else self.node.working_path / "venv"
+        )
+        cmd_result = super().run(
```

Comment:
@squirrelsc @LiliDeng  this super() does not give Python Tool. Is this expected behavior? 

![image](https://github.com/microsoft/lisa/assets/26188281/733661dd-b18a-4cfc-95dc-4cc7c9db4c46)

Reply:
Not sure why it doesn't work, but it should be. In this case, the `run()` is not overrides, so you can call `self.run()` too.

Reply:
self.run uses 'command' from PythonVenv class. Here I have used to get the command from Python class.
Similarly using 'super().command' explicitly as well.

Should I replace the usage with node.execute('python3 ....')?